js/noms/src/database.js

```diff
@@ -63,7 +65,30 @@ export default class Database {
    * id in the above Datasets Map.
    */
   getDataset(id: string): Dataset {
```

Comment:
How about something along the lines of:

```js
getDataset(id: string): Dataset {
  const getRef = async () => {
    const sets = await this.datasets();
    const valRef = sets.get(id);
    ...
  };
  return new Dataset(this, id, getRef());
}
```

all those then callbacks makes the code harder to follow.